### ISSUE TITLE
Bug: Project shows "Cluster not ready"

### ISSUE THREAD

**Issue by patrick0057** [rancher/rancher#27839]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Unknown, happened after an upgrade.  Unable to reproduce

**Result:**
![image](https://user-images.githubusercontent.com/44879867/86626076-07be0e00-bf7b-11ea-88ca-3e2bd44f0f9b.png)

**Other details that may be helpful:**
The issue is related to DefaultNetworkPolicyCreated in API being set to unknown.  
```
"lastUpdateTime": "2020-05-14T06:40:41Z",
"status": "Unknown",
"type": "DefaultNetworkPolicyCreated"
```

Previously it was possible to delete projectnetworkpolicies,cattle,io for the project in question and then force a sync by editing the project's description which would fix the unknown status.  In two other instances where this was seen however this didn't fix it.  Instead it came back as unknown.
```
kubectl delete projectnetworkpolicies.management.cattle.io -n p-69fs2 
```

At the moment the only workaround is to delete the Unknown status directly from the project which will fix the UI although the underlying bug still exists.
```
kubectl get projects.management.cattle.io -A
kubectl get projects.management.cattle.io -n local p-ldn9h -o yaml > backup.yaml
kubectl edit projects.management.cattle.io -n local p-ldn9h
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.4.3
- Installation option (single install/HA): HA

gz#11059
gz#11056
gz#11206
gz#10305
gz#12204
gz#13165
gz#13547

gz#14814

gz#14959

gz#15932

**Comment by patrick0057** [comment]
Additional information:
- The happened after an upgrade from Rancher v2.3.5 to v2.4.3 and upgrade from Kubernetes v1.15.9 to v1.17.5,  
- The issue was observed in the System project of a downstream cluster

**Comment by PhilipSchmid** [comment]
I've got the same problem with Rancher version `v2.4.8`. Would be awesome to see this actually fixed in `v2.4.9`.

**Comment by hcouplet** [comment]
I have the same issue upgrading from v2.4.5 to v2.4.8

**Comment by chrisgelhaus** [comment]
We've experienced the same issue in a Rancher managed cluster after upgrading to 2.4.8. To troubleshoot, I opened our Rancher console, selected the cluster with the issue, then clicked Launch kubectl

When we try the workaround, the first kubectl command gives the following error
kubectl get projects.management.cattle.io -A
error: the server doesn't have a resource type "projects"

I have confirmed it is not in the list by running this command:
kubectl get crd
networksets.crd.projectcalico.org             2020-06-04T19:11:58Z
prometheuses.monitoring.coreos.com            2020-06-04T19:25:53Z

This tells me neither of the next two commands would work either. Do I have the option to create the entry? If so, would it look like this? 
kubectl create crd -n local c-vfczk
And which cluster would I run it in? The Rancher cluster or the managed cluster with the issue?

**Comment by hcouplet** [comment]
same for me : 
![image](https://user-images.githubusercontent.com/11631430/94157180-37492c00-fe81-11ea-9ae7-157b2fcabc67.png)

If someone has a workaround, I am interested.
Currently I am holding upgrade on other clusters.

**Comment by chrisgelhaus** [comment]
@hcouplet , We are holding other cluster updates as well. Thank you for the screen shots. They mirror what we are seeing.
@patrick0057 , any thoughts?

**Comment by PhilipSchmid** [comment]
Hi @hcouplet and @chrisgelhaus 

The commands from @patrick0057 need to be applied on the `local` Rancher cluster itself (not on the Rancher managed ones). I was able to fix it using the commands down here:
```
kubectl get projects.management.cattle.io -A 
kubectl get projects.management.cattle.io -n c-rsabc p-8r123 -o yaml > ~/Temp/backup.yaml
# Remove the whole entry with status "unavailable":
kubectl edit projects.management.cattle.io -n c-rsabc p-8r123
```
For me the regarding project object was inside another namespace (`p-8r123`) and not inside `local` as @patrick0057 wrote.

Hope this helps! If not please comment again. I hope Rancher fixes this issue soon (we opened a ticket via their support).

Regards,
Philip

**Comment by hcouplet** [comment]
This worked for me. Thanks.

As I am running rancher as docker, I had to run commands after : 
`
docker exec -it rancher2 bash
`
![image](https://user-images.githubusercontent.com/11631430/95922253-28d09f00-0db3-11eb-9db1-e26e896e8097.png)

Then, used the /tmp/ directory : 
`
kubectl get projects.management.cattle.io -n c-vg59f p-ddhwb -o yaml > /tmp/backup.yaml
`
As vi was not in the docker image, I ran
`
apt-get update
apt install vim-tiny
`
![image](https://user-images.githubusercontent.com/11631430/95922425-92e94400-0db3-11eb-8983-6154f078d6ed.png)

**Comment by dnoland1** [comment]
Hitting this too:
```
  - lastUpdateTime: "2020-09-01T18:49:55Z"
    status: Unknown
    type: DefaultNetworkPolicyCreated
```

**Comment by livenson** [comment]
Hi, confirming, that deleting the section from the project description restored status to Active. For us it was also about DefaultNetworkPolicyCreated:

`  - lastUpdateTime: "2020-11-02T04:02:03Z"
    status: Unknown
    type: DefaultNetworkPolicyCreated`

**Comment by Napsty** [comment]
Confirming here too, this happened on a new installation after a Rancher version was downgraded from 2.4.15 to 2.4.5.

```
  status:
    conditions:
    - lastUpdateTime: "2021-05-06T19:28:10Z"
      status: "True"
      type: BackingNamespaceCreated
    - lastUpdateTime: "2021-05-06T19:28:10Z"
      status: "True"
      type: CreatorMadeOwner
    - lastUpdateTime: "2021-05-06T19:28:10Z"
      status: "True"
      type: InitialRolesPopulated
    - lastUpdateTime: "2021-05-06T19:33:36Z"
      status: "True"
      type: MetricExpressionDeployed
    - status: Unknown
      type: DefaultNetworkPolicyCreated
    podSecurityPolicyTemplateId: ""
```

The steps mentioned by @patrick0057 work to fix it and get the "System" project available in the UI again. Thanks for sharing this.

**Comment by Tejeev** [comment]
@Napsty I think this might be an effect of some of the issues that have been resolved in more recent versions (regarding our logic around communication with downstream clusters, iirc). Have you seen it on v2.4.15 or any other recent versions of Rancher?

**Comment by Napsty** [comment]
@Tejeev Actually no, out of all the Rancher installations this was the first time this happened. Could be related to the mentioned downgrade, I'm not sure about the exact cause.

**Comment by Jono-SUSE-Rancher** [comment]
@sowmyav27 - According to the recent chatter on this ticket, it looks like this might have been fixed as part of something else we did post 2.4x, can you guys do a quick verification on this?

**Comment by Jono-SUSE-Rancher** [comment]
@Tejeev - If possible, could you provide reproduction steps for the QA team? Sowmya worked on this and wasn't able to reproduce successful, so we are thinking recent changes might have closed this but we want to confirm.

**Comment by Tejeev** [comment]
@Jono-SUSE-Rancher sorry I have none.  One of our customers had this back in 2.4.3 but they since upgraded and have not seen any issues for a very long time

**Comment by alvaromlg** [comment]
@Jono-SUSE-Rancher Although is not a natural way to reproduce the issue you can edit the project medatada and modify the status to "error", "unkown", etc and it will reach the inconsistent state where "Cluster is not ready" but you can still access the project through the direct url.

On my case where the bug appeared when upgrading the monitoring app #33104 I could reproduce it in other projects by adding these lines:
```
  - lastUpdateTime: "2021-06-03T06:33:11Z"
    reason: Error
    status: "False"
    type: MonitoringEnabled
```
Although to reproduce it without manually editing the project template would require to manage having a failed monitoring app installation which would be tricky.

Maybe there are easier ways to get a failed status on project metadata.

**Comment by Jono-SUSE-Rancher** [comment]
QA is unable to reproduce this as well as us not seeing it on our current master branch. It looks like this will be fixed in v2.6. @Tejeev can you validate that this would be acceptable?
